docs/source/run/parameters.rst

```diff
@@ -504,25 +482,114 @@ which are valid only for certain beam types, are introduced further below under
 * ``<beam name> or beams.do_reset_id_init`` (`bool`) optional (default `0`)
     Whether to reset the ID incrementor to 1 before initializing beam particles.
 
-* ``<beam name> or beams.initialize_on_cpu`` (`bool`) optional (default `0`)
-    Whether to initialize the beam on the CPU instead of the GPU.
-    Initializing the beam on the CPU can be much slower but is necessary if the full beam does not fit into GPU memory.
+Option: ``fixed_weight_pdf``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* ``<beam name>.num_particles`` (`int`)
+    Number of constant weight particles to generate the beam.
+
+* ``<beam name>.pdf`` (`float`)
+    Probability density function to define the longitudinal profile of the beam
+    (the transverse profile is Gaussian). This is a parser function of z
+    that should return the total amount of charge present on each transverse beam slice.
+    This will also be proportional to the per-slice peak density if ``<beam name>.position_mean`` is constant along z,
+    if not the pdf can be calculated from the desired peak density using ``<beam name>.pdf = <peak_density(z)> * <position_std_x(z)> * <position_std_y(z)>``.
+    The probability density function is automatically normalized so that all constant pre factors
+    can be omitted when specifying the function.
+    Examples (replace `z_center`, `z_std`, `z_length`, `z_slope`, `z_min` and `z_max`):
+    Gaussian: ``exp(-0.5*((z-z_center)/z_std)^2)``
+    Cosine: ``(cos(2*pi*(z-z_center)/z_length)+1)*(2*abs(z-z_center)<z_length)``
+    Trapezoidal: ``(z<z_max)*(z>z_min)*(1+z_slope*z)``
+
+* ``<beam name>.total_charge`` (`float`)
+    Total charge of the beam. Note: Either ``total_charge`` or ``density`` must be specified.
+    Only available when running in SI units.
+    The absolute value of this parameter is used when initializing the beam.
+    Note that in contrast to the ``fixed_weight`` injection type, using ``<beam name>.radius`` or
+    a special pdf to emulate `z_min` and `z_max` will result in beam particles being redistributed to
+    other locations and not in them being deleted in the injection, meaning that the resulting beam will
+    have exactly the total charge that is specified here.
```

Comment:
```suggestion
    other locations rather than being deleted. Therefore, the resulting beam will have exactly the specified total charge, but cutting a significant fraction of the charge is not recommended.
```